go/proof.go

```diff
@@ -163,6 +168,95 @@ func (p *ExistenceProof) calculate(spec *ProofSpec) (CommitmentRoot, error) {
 	return res, nil
 }
 
+func (p *ExclusionProof) Calculate() (CommitmentRoot, error) {
+	if p.GetLeaf() == nil {
+		return nil, errors.New("exclusion proof needs defined LeafOp")
+	}
+	if p.Leaf.PrehashKey != HashOp_NO_HASH {
+		return nil, errors.New("exclusion proof must have leaf with PrehashKey == NO_HASH")
+	}
+	if p.Leaf.PrehashValue != HashOp_NO_HASH {
+		return nil, errors.New("exclusion proof must have leaf with PrehashValue == NO_HASH")
+	}
```

Comment:
Why is this a requirement? Consider adding a comment or updating the error message with the reasoning

Reply:
I left a comment explaining this in the proto but will add again here